docker/test/integration/minifi/core/DockerTestCluster.py

```diff
@@ -34,6 +34,15 @@ def get_stdout_encoding():
         return encoding
 
     def get_app_log(self, container_name):
+        type = self.containers[container_name].type()
+        if type == 'docker container':
+            return self.__get_app_log_from_docker_container(container_name)
+        elif type == 'direct':
+            return self.containers[container_name].get_app_log()
+        else:
+            raise Exception("Unexpected container type '%s'" % type)
```

Comment:
I would either change the method name from `type()` to `log_source()` or the type `direct` to `kind cluster container` or something similar to be more clearer, because `direct` doesn't really name a type. Also maybe a class could be added with const members for these literals to be used from a single place like

```
class ContainerType:
  DOCKER_CONTAINER = "docker container"
  KIND_CLUSTER_CONTAINER = "kind cluster container"
```

Reply:
Part of the problem is that we have our own `minifi.core.Container` (and derived classes), and the docker client's `docker.models.Container`, and we call both "container".

The difference here is whether we get the logs from the `docker.models.Container` contained in the `minifi.core.Container`, or directly from the `minifi.core.Container`, using its `get_app_log()` method.  This could be used more generally than Kubernetes or Kind, so "kind cluster container" is too specific.

Reply:
I have renamed `type()` to `log_source()` and created an enum-like class for the values as you suggested (https://github.com/apache/nifi-minifi-cpp/pull/1270/commits/4f416b5c2a955c7ed6e7af6aa7dbdb911e4ecccf).